pkg/couchdb/bulk.go

```diff
@@ -368,7 +371,47 @@ func BulkForceUpdateDocs(db prefixer.Prefixer, doctype string, docs []map[string
 		NewEdits: false,
 		Docs:     docs,
 	}
+	logBulk(db, "BulkForceUpdateDocs", doctype, docs)
 	// XXX CouchDB returns just an empty array when new_edits is false, so we
 	// ignore the response
 	return makeRequest(db, doctype, http.MethodPost, "_bulk_docs", body, nil)
 }
+
+func logBulk(db prefixer.Prefixer, prefix, doctype string, docs interface{}) {
+	extracted := make([]string, 0, 1000)
+	if documents, ok := docs.([]Doc); ok {
+		for _, doc := range documents {
+			id := doc.ID()
+			rev := revision.Generation(doc.Rev())
+			extracted = append(extracted, fmt.Sprintf("%s (%d)", id, rev))
+		}
+	} else if updates, ok := docs.([]UpdateResponse); ok {
+		for _, update := range updates {
+			id := update.ID
+			rev := revision.Generation(update.Rev)
+			extracted = append(extracted, fmt.Sprintf("%s (%d)", id, rev))
+		}
+	} else if maps, ok := docs.([]map[string]interface{}); ok {
+		for _, doc := range maps {
+			id, _ := doc["_id"].(string)
+			extracted = append(extracted, fmt.Sprintf("%s", id))
+		}
+	}
+
+	var messages []string
+	for len(extracted) > 0 {
+		nb := len(extracted)
+		// We limit the number of ids per log to avoid the line width limit.
+		if nb > 50 {
+			nb = 50
+		}
+		msg := strings.Join(extracted[:nb], " ")
```

Comment:
Maybe we could use another separator as there can be spaces in the extracted strings?!

Reply:
No, spaces is practical for searching in logs